### ISSUE TITLE
Shapely throws error on linemerge when numpy not present

### ISSUE THREAD

**Issue by grant-humphries** [Toblerity/Shapely#431]
**Environment details:**
Ubuntu 16.04
GEOS 3.5.0-1 (libgeos-dev)
Python 2.7.12 (python-dev)
Shapely 1.5.17

Hi folks,

Shapely is able to find GEOS and install (I'm using buildout to fetch python dependencies), but I don't have numpy and the below error message was thrown when I called linemerge.  I'd prefer not require numpy to run my code just to keep project dependencies minimal.

```
Traceback (most recent call last):
  File "./bin/rlis2osm", line 22, in <module>
    sys.exit(rlis2osm.main.main())
  File "/home/ubuntu/rlis2osm/rlis2osm/main.py", line 228, in main
    dissolver.dissolve_ways(paths.combined, paths.dissolved)
  File "/home/ubuntu/rlis2osm/rlis2osm/dissolve.py", line 40, in dissolve_ways
    dissolve_geom = linemerge(geom_list)
  File "/home/ubuntu/rlis2osm/eggs/Shapely-1.5.17-py2.7-linux-x86_64.egg/shapely/ops.py", line 113, in linemerge
    result = lgeos.GEOSLineMerge(source._geom)
  File "/home/ubuntu/rlis2osm/eggs/Shapely-1.5.17-py2.7-linux-x86_64.egg/shapely/geometry/proxy.py", line 33, in _geom
    self.__geom__, n = self.factory(self.context)
  File "/home/ubuntu/rlis2osm/eggs/Shapely-1.5.17-py2.7-linux-x86_64.egg/shapely/geometry/multilinestring.py", line 134, in geos_multilinestring_from_py
    geom, ndims = linestring.geos_linestring_from_py(obs[l])
  File "/home/ubuntu/rlis2osm/eggs/Shapely-1.5.17-py2.7-linux-x86_64.egg/shapely/geometry/linestring.py", line 197, in geos_linestring_from_py
    ob = required(ob)
  File "/home/ubuntu/rlis2osm/eggs/Shapely-1.5.17-py2.7-linux-x86_64.egg/shapely/coords.py", line 31, in required
    return numpy.require(ob, numpy.float64, ["C", "OWNDATA"])
NameError: global name 'numpy' is not defined
```

**Comment by snorfalorpagus** [comment]
I hadn't thought much about the numpy dependency as I use it so much for other projects.

Shapely *is* supposed to work without numpy. I think it's an oversight that it isn't tested on Travis; we do builds with/without cython, but not without numpy.

WRT to your project, you may want to accept the dependency. The cython speedups can't be built without numpy and they provide a significant increase in performance for certain operations.

**Comment by snorfalorpagus** [comment]
The actual changes required to get Shapely working without numpy are quite trivial (just a few lines). See build here: https://travis-ci.org/snorfalorpagus/Shapely/jobs/177257573

I haven't quite figured out how to prevent pip from pulling in numpy when it builds shapely. In the build above I just manually uninstall it before running the tests, which is a hack.

**Comment by sgillies** [comment]
I'm willing to look into Numpy-free installations of Shapely. Like @snorfalorpagus says, it's mostly an oversight that it's not possible now.

**Comment by grant-humphries** [comment]
Thanks for the responses guys.  I'm not really opposed to installing numpy, especially given the performance benefits that it brings, my only hesitation is that it can be a little difficult to install on some platforms.  But if it's a fairly easy fix having a numpy free option would be nice.

**Comment by snorfalorpagus** [comment]
Closed by #433.
